crates/turbopack-dev-server/src/source/resolve.rs

```diff
@@ -83,26 +72,55 @@ pub async fn resolve_source_request(
                                 );
                             }
                         }
-                        current_asset_path = new_asset_path;
-                        data = ContentSourceData::default();
-                    }
-                    ContentSourceContent::NotFound => {
-                        break Ok(ResolveSourceRequestResult::NotFound.cell());
+                        // do the rewrite
+                        match &rewrite.ty {
+                            RewriteType::Location { path_and_query } => {
+                                let new_uri = Uri::try_from(path_and_query)?;
+                                let new_asset_path =
+                                    urlencoding::decode(&new_uri.path()[1..])?.into_owned();
+                                request_overwrites.uri = new_uri;
+                                current_asset_path = new_asset_path;
+                                continue 'routes;
+                            }
+                            RewriteType::ContentSource {
+                                source,
+                                path_and_query,
+                            } => {
```

Comment:
I this this rewrite logic creates a lot of inconsistencies. Before we were rewriting the current sources's URL only, but now we're changing the URL for the current route _and all routes that follow_. I think it has to only rewrite the the URL for the current route tree, right?

Reply:
If referring to the next route in `sources`, on `Rewrite` we do no longer apply following routes. The `continue 'routes;` will skip all remaining routes and starts applying a new route tree with the new URL.

The Rewrite is basically counted as route match.